plugins/modules/identity/keycloak/keycloak_client_scopemapping.py

```diff
@@ -0,0 +1,333 @@
+
+short_description: Administration of Keycloak client-level roles of the client's scope via the Keycloak API
+
+version_added: 4.4.0
+
+description:
+    - This module allows you to add, remove Keycloak client-level roles scope mapping with the Keycloak REST API.
+      It requires access to the REST API via OpenID Connect; the user connecting and the client being
+      used must have the requisite access rights. In a default Keycloak installation, admin-cli
+      and an admin user would work, as would a separate client definition with the scope tailored
+      to your needs and a user having the expected roles.
+
+    - The names of module options are snake_cased versions of the camelCase ones found in the
+      Keycloak API and its documentation at U(https://www.keycloak.org/docs-api/8.0/rest-api/index.html).
+
+    - When updating a scope mapping, where possible provide the role ID to the module. This removes a lookup
+      to the API to translate the name into the role ID.
+
+options:
+    state:
+        description:
+            - State of the client-level role scope mapping.
+            - On C(present), the client-level role scope mapping will be created if it does not yet exist, or updated with the parameters you provide.
+            - On C(absent), the client-level role scope mapping will be removed if it exists.
+        default: 'present'
+        type: str
+        choices:
+            - present
+            - absent
+    realm:
+        type: str
+        description:
+            - They Keycloak realm under which this role representation resides.
+        default: 'master'
+    client_id:
+        type: str
+        description:
+            - Name of the client to be mapped (different than I(cid)).
+            - This parameter is required (can be replaced by cid for less API call).
+    id:
+        type: str
+        description:
+            - ID of the client to be mapped.
+            - This parameter is not required for updating or deleting the rolemapping but
+              providing it will reduce the number of API calls required.
+    client_role:
+        description: Scope mapping of the client-level role.
+        type: dict
+        suboptions:
+            id:
+                description: ID of the client
+                type: str
+            client_id:
+                description: Name of the client
+                type: str
+            roles:
+                description: Client roles
+                type: list
+                elements: dict
+                suboptions:
+                    name:
+                        description: Name of the role
+                        type: str
+                    id:
+                        description: ID of the role
+                        type: str
+
+extends_documentation_fragment:
+- community.general.keycloak
+
+
+author:
+    - Fynn Chen (@fynncfchen)
+'''
+
+EXAMPLES = '''
+- name: Map a client role to a client, authentication with credentials
```

Comment:
Based on the parameters, it seems that this should instead say `Map several client roles`?

I haven't caught up with what this PR does yet, (need to test it to understand it better) so I might be mis-guided with this comment.

Reply:
You're right! I'll update the description, thanks!